src/celeritas/track/TrackInitParams.hh

```diff
@@ -18,6 +18,9 @@ namespace celeritas
 //---------------------------------------------------------------------------//
 /*!
  * Manage persistent track initializer data.
+ *
+ * \todo \c max_events could potentially be per thread, not global? And we
+ * should differentiate between user events and events in flight.
```

Comment:
Maybe? I think it's only used for the track counters, so having a max events per thread would make mapping to the event ID more complicated (unless I'm misunderstanding the comment).

Reply:
My thought is that it's nonsensical to use `max_events` in a multi-stream context in the likely case that different streams have different event ids.

Reply:
Trying to recall what our plan was for the event IDs/counters:
- In Geant4-integrated apps, `max_events` is always set to one since #1447 
  - Currently the internal event ID is always set to zero and tracks IDs are sequential from zero up to the total number of tracks transported on that _stream_, not in each event
  - The goal is to have a single global counter for the internal event IDs (which will map to the user event IDs) and a single stream-local track counter that will be reset at the beginning of each event
- celer-sim event-parallel mode still sets `max_events` equal to the total number of events
- celer-sim combined-event mode will sill need `max_events` track counters

Reply:
I think that's all exactly right. I've updated https://github.com/celeritas-project/celeritas/issues/1233 to try to better explain the problems and potential avenues of research. For `celer-sim` we don't (yet) care about actual event IDs so we can just use it to measure performance of multiple simultaneous streams.